File: src/main/java/dev/gustavpersson/thorincompanies/business_logic_layer/models/Company.java
package dev.gustavpersson.thorincompanies.business_logic_layer.models;

import java.util.Date;
import java.util.UUID;

public class Company {

    public int id;

    //The display name for the company
    public String name;

    //The original founder
    public UUID founderUUID;
    private Date createdAt;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public UUID getFounderUUID() {
        return founderUUID;
    }

    public void setFounderUUID(UUID founderUUID) {
        this.founderUUID = founderUUID;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

}

File: src/main/java/dev/gustavpersson/thorincompanies/business_logic_layer/models/Company.kt
package dev.gustavpersson.thorincompanies.business_logic_layer.models

import java.math.BigDecimal
import java.time.LocalDate
import java.util.*

data class Company(
    val id: Int,
    val name: String,
    val founderUUID: UUID,
    val startupCapital: BigDecimal,
    val createdAt: LocalDate
)

data class NewCompany(
    val name: String,
    val founderUUID: UUID,
    val startupCapital: BigDecimal,
    val createdAt: LocalDate
)

data class RenameCompanyRequest(
    val id: Int,
    val name: String
)
